File: leetcode/python/0392_is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == t or len(s) == 0: return True
        if len(s) > len(t): return False

        i = 0
        for j in range(0, len(t)):
            if s[i] == t[j]: i += 1
            if i == len(s): return True

        return i == len(s)

File: leetcode/java/0392_is_subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0 && t.length() == 0 || (s.length() == 0)) return true;

        for (int i = 0, j = 0; i < s.length() && j < t.length(); ) {
            // if s[i] == j[i]: i++, j++
            // if s[i] != j[i]: j++
            // if i == s.length(): return true
            // return false after loop ends

            if (s.charAt(i) == t.charAt(j)) {
                i++;
                j++;
            } else {
                j++;
            }
            if (i == s.length()) return true;
        }

        return false;
    }
}